llvm/lib/Target/X86/X86ISelDAGToDAG.cpp

```diff
@@ -5271,6 +5271,52 @@ void X86DAGToDAGISel::Select(SDNode *Node) {
     }
     break;
   }
+  case ISD::FCANONICALIZE: {
+    SDValue Operand = Node->getOperand(0);
+    EVT VT = Node->getValueType(0);
+
+    // Perform canonicalization for constants. Replace the operand by a load
```

Comment:
The constant case shouldn't really need special casing in selection, especially if you are constant folding it elsewhere